### ISSUE TITLE
Unable to merge numeric labels

### ISSUE THREAD

**Issue by d9705996** [grafana/grafana#26641]
**What happened**:

When applying a [merge transform](https://grafana.com/docs/grafana/latest/panels/transformations/#merge) to a table panel using a prometheus data-source, series are not merged when there is a numeric label (Only a merge transform has been applied in the examples below)

![image](https://user-images.githubusercontent.com/2470279/88629931-cc10f280-d0a7-11ea-9af0-f16b77e96ff7.png)

**What you expected to happen**:

Series to be merged correctly.  As a workaround numeric labels can be forced to non-numeric using `label_replace` which causes the data to be merged correctly

![image](https://user-images.githubusercontent.com/2470279/88630301-493c6780-d0a8-11ea-8051-1ea4f4613491.png)

**Anything else we need to know?**:

I think the issue is related to the [numericComparer](https://github.com/grafana/grafana/blob/093383eb83a8166a1d044ebb41940366e83091f0/packages/grafana-data/src/field/fieldComparers.ts#L45-L47) function 

**Environment**:
- Grafana version: 7.1.0 & 7.1.1
- Data source type & version: prometheus
- OS Grafana is installed on: CentOS 7.6
- User OS & Browser: Windows 10, Chrome/Firefox
- Grafana plugins: Table
- Others:

**Comment by royantman** [comment]
is there any known workaround?

**Comment by royantman** [comment]
Answering myself to whomever needs this:

Just use **label_replace()** to add some character to the label value so it is considered as string.

Lets say **partition** label is considered as numeric.
```
sum by (topic, partition) (label_replace(foo{}, "partition", "p-$1", "partition", "(.*)"))
```

Ultra disgusting but didn't find anything else.

**Comment by aocenas** [comment]
Removing the Prom label as it does not seem Prom related that much and remove it from our internal issue triage pipeline.

**Comment by hugohaggmark** [comment]
@d9705996  I can't repro this issue after Grafana version > 7.2.0-beta1 and I think this PR might have fixed this #27362. Closing this issue for now, but we can always reopen it if we're presented with more evidence.
